examples/with-strict-csp/src/middleware.ts

```diff
@@ -0,0 +1,31 @@
+import { createMiddleware } from "@solidjs/start/middleware";
+import { randomBytes } from "crypto";
+
+const isProd = import.meta.env.PROD;
+
+export default createMiddleware({
+  onRequest: event => {
+    const nonce = randomBytes(16).toString("base64");
+
+    event.locals.nonce = nonce;
+
+    // Notes:
+    // 1. SolidStart uses `eval` for data serialization, which may require you to add 'unsafe-eval' to your CSP.
+    //    For more information, see: https://github.com/solidjs/solid-start/issues/1825
+    // 2. In development, Vite inlines small CSS files to enhance performance, so you will need to include 'unsafe-inline' in development mode.
+    // 3. During the build process, Vite inlines small assets as data URLs. Therefore, it's necessary to add data: to the relevant directives (e.g., img-src, font-src, etc.)
+    //    For more details, see: https://vite.dev/config/build-options.html#build-assetsinlinelimit
```

Comment:
I added this comment to clarify some of the confusing directives I used. I'm specifically unsure about point two. I couldn't find the documentation regarding Vite inlining small CSS files, but apparently, it happens during development. I encountered some errors related to this while testing.

Reply:
yes, it does. I have found that to be the required as well.
I think it's the critical CSS path 👍